sklearn/compose/tests/test_column_transformer.py

```diff
@@ -310,6 +310,50 @@ def test_column_transformer_empty_columns(pandas, column_selection,
     assert isinstance(ct.transformers_[0][1], TransRaise)
 
 
+def test_column_transformer_transformers_output():
+    X_array = np.array([[0, 1, 2], [2, 4, 6]]).T
+
+    ct = ColumnTransformer([('trans1', Trans(), [0]),
+                            ('trans2', Trans(), [1])])
+    ct.fit(X_array)
+    assert ct.transformers_output_ == {'trans1': slice(0, 1),
+                                       'trans2': slice(1, 2)}
+
+    # test with transformer_weights and multiple columns
+    ct = ColumnTransformer([('trans', Trans(), [0, 1])],
+                           transformer_weights={'trans': .1})
+    ct.fit(X_array)
+    assert ct.transformers_output_ == {'trans': slice(0, 2)}
+
+    # test case that ensures that the attribute does also work when
+    # a given transformer doesn't have any columns to work on
+    ct = ColumnTransformer([('trans1', Trans(), [0, 1]),
+                            ('trans2', TransRaise(), [])])
+    ct.fit(X_array)
+    assert ct.transformers_output_ == {'trans1': slice(0, 2)}
```

Comment:
Should `trans2` be assigned an empty slice then? I think that a KeyError would be unexpected.